### DISCUSSION TITLE
How do I get email sender to work properly?

### DISCUSSION THREAD

**Discussion by motoyugota** [OrchardCMS/OrchardCore#7458] [Q&A]
I am trying to get a workflow set up that sends an email from a contact form. I have it set up to use Gmail SMTP, with my email account entered for credentials. This works to send the email, but no matter what I put in for "Sender" (either in the settings, in the email test form, or in the Send Email task's From/Sender fields), it is always coming from my gmail account, not what I set the sender to.

I can't figure out what is going wrong here - it looks like the email code should be reading the values back from the test form fine. Is this just some problem with using gmail SMTP that I'm not aware of? I've used it in the past for other .net apps to send emails like this without issue.

**Comment by hishamco** [comment]
@motoyugota just one question did you try this with other mail provider?

**Reply by Skrypt** [reply]
I think you want to look at the ReplyTo option in that case.

**Reply by hishamco** [reply]
As I remember from more than 10 years whenever you send an email the email is sent from the email that is used in credentials, may be it's a limitation on some mail providers or libraries, but I need to verify this

Unit test may be needed here if it's not exist

**Reply by Skrypt** [reply]
Generally if it's sent by a email server it uses the email credentials and it can't be changed. All you can do is change the ReplyTo email else we could be able to send emails on behalf of anyone ...

**Reply by hishamco** [reply]
But why the APIs offers a sender as setter property, I will dig into that ...

**Reply by motoyugota** [reply]
The replyto option is not available in the Send Email workflow task. It has the following:

From
The author's email address that defines who the email is from. Leave blank to use the configured email address. With Liquid support.

Sender
The sender's email address. Required if not the same as the From email address. With Liquid support.

The task's code (at RC2, which is what I have running) does appear to use these two fields:

```
            var message = new MailMessage
            {
                // Author and Sender are both not required fields.
                From = author?.Trim() ?? sender?.Trim(),
                To = recipients.Trim(),
                Subject = subject.Trim(),
                Body = body?.Trim(),
                IsBodyHtml = IsBodyHtml
            };
```

However, it appears that Google no longer allows you to use a different email address to send through their servers. I know I've done this in the past, but a simple test shows it no longer supports that, and it sounds like pretty much all mail services are the same at this point as well, so people can't spoof emails (which is good, I guess, even if it's a pain for situations like this).

Looking at the latest code, there is an update that does include the reply-to, so maybe I'll need to pull that in, but I'm thinking it should still be clear that those other fields don't really do anything unless you've got your own mail server which you can still set up to spoof email addresses.

**Reply by hishamco** [reply]
I will check from my side too
